k2/csrc/properties.cc

```diff
@@ -1,36 +1,57 @@
 // k2/csrc/properties.cc
 
-// Copyright (c)  2020  Daniel Povey
+// Copyright (c)  2020 Haowen Qiu
+//                     Daniel Povey
 
 // See ../../LICENSE for clarification regarding multiple authors
 
-#include "k2/csrc/properties.h"
+#include <unordered_set>
 
+#include "k2/csrc/properties.h"
```

Comment:
Since this file is `properties.cc`, so `properties.h` should be the ****first**** include header.

It should be placed before 
`#include <unordered_set>`

Reply:
ok.. but I really don't like this rule from google style...

Reply:
this is also the coding style for LLVM.

Reply:
Here is the motivation to always, **always** include it first.

https://llvm.org/docs/CodingStandards.html#include-style

After reading this, I think you will follow the style.